### PR TITLE
[REF] mail: rename CannedResponse and add to store

### PR DESCRIPTION
**Pull request by seb-odoo**
Support for delete is added in `Store` / `store.insert()`.

Part of task-3605717

### CODE REVIEW

File: addons/mail/static/tests/composer/composer.test.js
Comment (seb-odoo) [inline addons/mail/static/tests/composer/composer.test.js:1]: Note: the test didn't need diff to pass, I'm just making it more robust/maintainable as I had issues with it during dev.

File: addons/mail/static/tests/composer/composer.test.js [hunk 1]
```diff
@@ -1022,15 +1022,16 @@ test("Canned response can be deleted from the bus", async () => {
     ]);
     await start();
     await openDiscuss(channelId);
-    await contains(".o-mail-Composer-input");
     await insertText(".o-mail-Composer-input", ":");
     await contains(".o-mail-NavigableList-item", { count: 2 });
+    await contains(".o-mail-NavigableList-item", { text: "hello" });
```
Comment (seb-odoo) [inline addons/mail/static/tests/composer/composer.test.js:1027]: Note: the test didn't need diff to pass, I'm just making it more robust/maintainable as I had issues with it during dev.

**Review by alexkuhn — APPROVED** [review]
@robodoo delegate+

**Review by seb-odoo — COMMENTED** [review]
@robodoo r+

**Pull request closed without merging**